datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -319,13 +319,20 @@ fn date_bin_single(stride: i64, source: i64, origin: i64) -> i64 {
 
 /// DATE_BIN sql function
 pub fn date_bin(args: &[ColumnarValue]) -> Result<ColumnarValue> {
-    if args.len() != 3 {
+    if args.len() != 2 && args.len() != 3 {
```

Comment:
Thanks, unfortunately We poorly support overloaded functions, and have to use confusing conditions.

However maybe it can be improved like 

```
pub fn date_bin(args: &[ColumnarValue]) -> Result<ColumnarValue> {
    check params count
    if 3 the date_bin_3args
    if 2 the date_bin_2args
    else panic

}

pub fn date_bin_3args(args: &[ColumnarValue]) -> Result<ColumnarValue> {
    main logic
}

pub fn date_bin_2args(args: &[ColumnarValue]) -> Result<ColumnarValue> {
    order params
    date_bin_3args(params);
}
```

Reply:
How could we improve this limitation? Maybe open a new ticket for it.

Reply:
Here is one proposed solution: https://github.com/apache/arrow-datafusion/pull/5673